dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -87,12 +87,31 @@ check_not_exporting_variables() {
     fi
 }
 
+# Deprecated variables or patterns
+# Prevent deprecated variables or patterns to re-appear after cleanup
+check_deprecated_vars_patterns() {
+    local patterns=()
+    local pathspec=()
+
+    patterns+=(-e 'FEATURES_MCU_GROUP')
+
+    # Pathspec with exclude should start by an inclusive pathspec in git 2.7.4
+    pathspec+=('*')
+
+    # Ignore this file when matching as it self matches
+    pathspec+=(":!${SCRIPT_PATH}")
+
+    git -C "${RIOTBASE}" grep "${patterns[@]}" -- "${pathspec[@]}" \
+        | sed '1i \\nDeprecated variables or patterns:'
```

Comment:
@aabadie #11672 was merged and changed the way to do the "buildsystem_sanity_check". Can you please rebase and adapt ?
```suggestion
        | error_with_message 'Deprecated variables or patterns:'
```